src/commands/createNewProject/ProjectCreateStep/JavaScriptProjectCreateStep.ts

```diff
@@ -50,8 +45,47 @@ export class JavaScriptProjectCreateStep extends ScriptProjectCreateStep {
         }
     }
 
-    protected getPackageJsonDevDeps(_context: IProjectWizardContext): { [key: string]: string } {
-        return {};
+    protected getPackageJson(context: IProjectWizardContext): { [key: string]: unknown } {
+        const packageJson: { [key: string]: unknown } = {
+            name: convertToValidPackageName(path.basename(context.projectPath)),
+            version: '1.0.0',
+            description: '',
+            scripts: this.getPackageJsonScripts(context),
+            dependencies: this.getPackageJsonDeps(context),
+            devDependencies: this.getPackageJsonDevDeps(context)
+        };
+
+        if (isNodeV4Plus(context)) {
+            // default functionSubpath value is a string
+            const functionSubpath: string = getWorkspaceSetting(functionSubpathSetting) as string;
+            packageJson.main = path.posix.join(functionSubpath, '*.js');
+        }
+
+        return packageJson;
+    }
+
+    protected getPackageJsonScripts(_context: IProjectWizardContext): { [key: string]: string } {
+        return {
+            start: 'func start',
+            test: 'echo \"No tests yet...\"'
+        }
+    }
+
+    protected getPackageJsonDeps(context: IProjectWizardContext): { [key: string]: string } {
+        const deps: { [key: string]: string } = {};
+        if (isNodeV4Plus(context))
+            deps[azureFunctionsDependency] = azureFunctionsDependencyVersion;
+
+        return deps;
+    }
+
+    protected getPackageJsonDevDeps(context: IProjectWizardContext): { [key: string]: string } {
+        const devDeps: { [key: string]: string } = {};
+        if (isNodeV4Plus(context)) {
+            devDeps['@types/node'] = '^18.0.0';
```

Comment:
This is only needed for TypeScript projects, not JavaScript

Reply:
Would `^18.x` work for the node version for TypeScript?

Reply:
Yeah I think `^18.x` and `^18.0.0` are pretty much the same.